pkg/common/common.go

```diff
@@ -484,3 +487,41 @@ func GetVolumeMode(volUtil util.VolumeUtil, fullPath string) (v1.PersistentVolum
 	}
 	return "", fmt.Errorf("Block device check for %q failed: %s", fullPath, errblk)
 }
+
+// NodeExists checks to see if a Node exists in the Indexer of a NodeLister.
+func NodeExists(nodeLister corelisters.NodeLister, nodeName string) (bool, error) {
+	_, err := nodeLister.Get(nodeName)
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	return err == nil, err
+}
+
+// NodeAttachedToLocalPV gets the name of the Node that a local PV has a NodeAffinity to.
+// It assumes that there should be only one matching Node for a local PV and that
+// the local PV follows the form:
+//
+//	nodeAffinity:
+//	  required:
+//	    nodeSelectorTerms:
+//	    - matchExpressions:
+//	      - key: kubernetes.io/hostname
+//	        operator: In
+//	        values:
+//	        - <node1>
+func NodeAttachedToLocalPV(pv *v1.PersistentVolume) (string, bool) {
+	nodeNames := volumeUtil.GetLocalPersistentVolumeNodeNames(pv)
+	if nodeNames == nil || len(nodeNames) == 0 {
+		return "", false
+	}
+	// We assume that there should only be one matching node.
+	return nodeNames[0], true
```

Comment:
I think if there are multiple node names, then it may not be safe to delete the pvc until all nodes are deleted. I this is a corner case we should not support initially.

Reply:
Changed to return false if there isn't exactly one node name and added unit test.